packages/@webex/plugin-cc/test/unit/spec/AgentConfigService/AgentConfigService.ts

```diff
@@ -0,0 +1,194 @@
+import { WebexSDK } from '../../../../src/types';
+import HttpRequest from '../../../../src/HttpRequest';
+import AgentConfigService from '../../../../src/AgentConfigService/AgentConfigService';
+
+jest.mock('../../../../src/HttpRequest', () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      request: jest.fn(),
+    };
+  });
+});
+
+describe('AgentConfigService', () => {
+  let agentConfigService: AgentConfigService;
+  let mockHttpRequest: jest.Mocked<HttpRequest>;
+  const mockWebexSDK: WebexSDK = {
+    logger: {
+      log: jest.fn(),
+    },
+  } as unknown as WebexSDK;
+  const mockWccAPIURL = 'https://api.example.com/';
+  const mockAgentId = 'agent123';
+  const mockOrgId = 'org123';
+
+  beforeEach(() => {
+    mockHttpRequest = new HttpRequest(mockWebexSDK) as jest.Mocked<HttpRequest>;
+    agentConfigService = new AgentConfigService(mockAgentId, mockOrgId, mockWebexSDK, mockWccAPIURL);
+    agentConfigService.requestInstance = mockHttpRequest;
```

Comment:
Won't the HttpRequest instance be created already as part of AgentConfigService instantiation? We can just use the actual instance and use it to return resolved and rejected values. And spy on it to check for request method invocation as well

Reply:
Taken Care!